keras_cv/models/object_detection/faster_rcnn.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+from keras_cv.bounding_box.converters import _decode
+from keras_cv.layers.object_detection.anchor_generator import AnchorGenerator
+from keras_cv.layers.object_detection.roi_align import _ROIAligner
+from keras_cv.layers.object_detection.roi_generator import ROIGenerator
+from keras_cv.layers.object_detection.roi_sampler import _ROISampler
+from keras_cv.layers.object_detection.rpn_label_encoder import _RpnLabelEncoder
+from keras_cv.ops.box_matcher import ArgmaxBoxMatcher
+
+
+def _resnet50_backbone():
+    inputs = tf.keras.layers.Input(shape=(None, None, 3))
+    x = inputs
```

Comment:
we should support an `include_rescaling` argument in pre-configured backbones.  You can do this by adding `include_resdcaling` to _resnet50_backbone(include_rescaling).

Then inside of the constructor,

```
if include_rescaling:
    x = tf.keras.applications.resnet.preprocess_inputs(x)
```

This allows us to make a contract to users going forwards that if they want to feed [0, 255] range images to models it will work with default the optimizer and loss scaling.

Reply:
Sounds good. Done.